content/doc/pipeline/index.adoc

```diff
@@ -1,981 +1,437 @@
 
-```groovy
-currentBuild.result
-currentBuild.displayName
-currentBuild.description
-```
+To add one or more steps from Snippet Generator to your pipeline code:
 
-## Existing Jobs
-Existing jobs can be triggered from your pipeline via the build command (eg: ```build 'existing-freestyle-job'```). You can also pass parameters to your external jobs as follows:
+. Open Snippet Generator
+. Scroll to the step you want
+. Click that step
+. Configure the selected step, if presented with configuration options
+. Click *Generate Groovy* to see a Groovy snippet that runs the step as configured
+. Optionally select and configure additional steps
 
-```groovy
-def job = build job: 'say-hello', parameters: [[$class: 'StringParameterValue', name: 'who', value: 'DZone Readers']]
-```
+image::/images/pipeline/snippet-generator.png[title="Snippet Generator", 800]
 
-# Controlling Flow
+When you click *Generate Groovy* after selecting a step, you see the function name used for that step, the names of any parameters it takes (if they are not default parameters), and the syntax used by Snippet Generator to create that step. 
 
-Because Jenkins Pipeline is based on the Groovy language, there are many powerful flow control mechanisms familiar to developers and operations teams, alike. In addition to standard Groovy flow control mechanisms like if statements, try/catch, and closures, there are several flow control elements specific to Pipeline.
+You can copy and paste the generated code right into your Pipeline, or use it as a starting point, perhaps deleting any optional parameters that you do not need.
 
-## Handling Approvals
+To access information about steps marked with the help icon (question mark), click on that icon.
 
-Jenkins Pipeline supports approvals, manual or automated, through the input step:
+== Basic Groovy Syntax for Pipeline Configuration
 
-```groovy
-input 'Are you sure?'
-```
+You typically add functionality to a new pipeline by performing the following tasks:
 
-With the submitter parameter, the input step integrates Jenkins security system to restrict the allowed approvers.
+* Adding nodes
+* Adding more complex logic (usually expressed as stages and steps)
+* Using the “ws” step to create additional workspace on an agent without taking another executor slot
 
-The input step in Jenkins Pipeline Stage View UI:
+To configure a pipeline you have created through the Jenkins UI, select the pipeline and click *Configure*.
 
-![Input Step](images/4-input-ui-hr.png)
+If you run Jenkins on Linux or another Unix-like operating system with a Git repository that you want to test, for example, you can do that with syntax like the following, substituting your own name for “joe-user”:
 
-## Timing
+ node {
+     git url: 'https://github.com/joe_user/simple-maven-project-with-tests.git'
+     def mvnHome = tool 'M3'
```

Comment:
I think this is a poor example to cite here, this requires that the Jenkins already has a Maven tool installer configured and named "M3" in the "Manage Jenkins" dialog.

Perhaps just pretending our fake project here uses the [maven wrapper](https://github.com/takari/maven-wrapper) (or Gradle!)

```groovy
node {
  git url: 'https://github.com/joe_user/simple-maven-project-with-tests.git'
  sh './mvnw -B verify'
```